src/Externs.cs

```diff
@@ -64,7 +64,11 @@ public partial class __default {
     }
 
     private static Double BigRationalToDouble(BigRational r) {
-      return (double)r.num / (double)r.den;
+      if (r.num == 0) {
+        return 0.0;
+      } else {
+        return (double)r.num / (double)r.den;
+      }
```

Comment:
As a follow-up after this PR, we should add this method to `BigRational` itself, so any other future code that needs to do this doesn't fall into the same trap.

Reply:
I originally had code in that library do do this, but ultimately removed it because it's one of several possible approximations. The rationale at the time was that the client of the library should have to explicitly choose to approximate, and how to approximate, because it's approximation isn't sound, strictly speaking. This issue makes me wonder whether that was the right choice, though! Perhaps a sufficiently scary name would be sufficient.

Reply:
Yeah I sympathize with that. It does seem like `ToDouble()` couldn't possibly have any other reasonable definition, but it is worth emphasizing that it's a lossy transformation.

Reply:
This definition is fine for here, but if we move it to the library there *is* actually a better one: we can find the closes binary fraction approximation of the bigrational, and then convert that to a double.  This matters when the numerator and denominator are very large but the result fits in a double: converting both sides to doubles first will loose a lot of precision even before performing the division.